src/MSALWrapper/ExceptionExtensions.cs

```diff
@@ -66,4 +67,29 @@ public static IEnumerable<Exception> GetAllExceptions(this Exception exception)
             }
         }
     }
+
+    /// <summary>
+    /// Extracts correlation IDs from the list of exceptions.
+    /// </summary>
+    /// <param name="exceptions">List of exceptions from which correlation IDs are extracted.</param>
+    /// <returns>List of correlation IDs.</returns>
+    public static List<string> ExtractCorrelationIDsFromException(IList<Exception> exceptions)

```

Comment:
Discussed offline - move this method into ExceptionListToStringConverter since it's not a true _extension_ method to the Exception class.